### PR TITLE
feat: unify framework runtime, CLI, transform pipeline

### PR DESCRIPTION
**Pull request by amadeuszsz**
## Summary

This PR consolidates the framework and runtime foundations around the current supported workflow, with calibration-status classification as the primary reference path.

- unify the CLI surface around `train`, `test`, `deploy`, `mlflow`, `session`, and `create-dataset`, including stronger completion coverage and consistent command dispatch
- restructure runtime code so CLI-facing entrypoints stay in `autoware_ml/scripts`, while reusable logic is moved into dedicated utility modules for deployment, MLflow, sessions, checkpoints, runtime setup, and dataset generation
- refactor the dataset and transform contract to a metadata-first flow:
  - datasets return sample metadata through `get_data_info()`
  - file loading happens explicitly in transforms
  - `PipelineContext` is introduced and propagated through transform composition
- clean up transform package structure and public APIs:
  - remove monolithic and wrapper-only modules
  - split camera transforms into focused modules
  - move generic point-cloud transforms out of the lidar namespace
  - keep package `__init__.py` files as the public import boundary
- align the calibration-status pipeline with the new framework contract:
  - rename the T4 config to `resnet18_t4dataset_j6gen2`
  - centralize calibration types
  - simplify dataset outputs around `calibration_data`
  - update calibration-status docs and coverage
- improve runtime behavior and experiment handling:
  - add the `test` command
  - add tmux-backed `session` commands for long-running jobs
  - add MLflow experiment export support and clean up MLflow entrypoints
  - make deployment logging and helper structure more consistent
- add shared optimizer and scheduler helpers to keep model runtime configuration explicit and testable, including support for model-defined optimizer groups and Hydra-partial materialization
- modernize framework typing to Python 3.11 style:
  - adopt PEP 585 built-in and `collections.abc` generics instead of deprecated `typing.List` / `typing.Dict` / `typing.Sequence` style aliases
  - adopt PEP 604 union syntax such as `X | None` instead of `Optional[X]`
  - drop redundant quoted custom-class annotations in files already using `from __future__ import annotations`
- refresh contributor and user docs to match the new package structure, command set, config naming, and run directory layout
- expand unit coverage for CLI behavior, sessions, MLflow utilities, deployment helpers, optimizer helpers, datamodule base behavior, calibration-status datasets, and renamed point-cloud transforms

## Checklist

- [ ] You've checked our [contribution overview](https://tier4.github.io/autoware-ml/contributing/contribution-overview/).
- [ ] Your PR follows our [pull request guidelines](https://tier4.github.io/autoware-ml/contributing/pull-request-guidelines/).

### CODE REVIEW

File: autoware_ml/transforms/camera/resize.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        height, width = image.shape[:2]
+        crop_center_noise_h = self._signed_random(0, self.crop_ratio / 2)
+        crop_center_noise_w = self._signed_random(0, self.crop_ratio / 2)
+        crop_center = np.array(
+            [height * (1 + crop_center_noise_h) / 2, width * (1 + crop_center_noise_w) / 2]
+        )
+
+        max_noise = max(abs(crop_center_noise_h), abs(crop_center_noise_w))
+        scale_noise = np.random.uniform(self.crop_ratio, 1 - max_noise)
+        scaled_h, scaled_w = height * scale_noise, width * scale_noise
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/transforms/camera/resize.py:39]: https://github.com/tier4/autoware-ml/pull/23/commits/a2e88b29ce2f0c4472e481513855fa8a8a800308

File: autoware_ml/tools/dataset/nuscenes/tasks/segmentation3d.py [hunk 2]
```diff
@@ -43,8 +52,8 @@ def process_sample(
         """
         if "lidarseg" in nusc.table_names:
             lidar_token = sample["data"]["LIDAR_TOP"]
-            info_dict["pts_semantic_mask_path"] = osp.join(
-                nusc.dataroot, nusc.get("lidarseg", lidar_token)["filename"]
+            info_dict["pts_semantic_mask_path"] = osp.basename(
+                nusc.get("lidarseg", lidar_token)["filename"]
             )
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/tools/dataset/nuscenes/tasks/segmentation3d.py:55]: https://github.com/tier4/autoware-ml/pull/23/commits/b4025e4bdf29343c16cb7e6ac4fb980ba183b5df

File: docs/user-guide/mlflow.md [hunk 3]
```diff
@@ -71,7 +107,7 @@ autoware-ml train --config-name my_task/my_model/my_config \
 import mlflow
 
 mlflow.set_tracking_uri("sqlite:///mlruns/mlflow.db")
-experiment = mlflow.get_experiment_by_name("tasks_my_task_my_model")
+experiment = mlflow.get_experiment_by_name("<task>/<model>/<config>")
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply docs/user-guide/mlflow.md:111]: https://github.com/tier4/autoware-ml/pull/23/commits/73724d8c2d7c50a5fffce6cb6685456361a4c51c

File: docs/framework/design.md [hunk 4]
```diff
@@ -128,22 +128,30 @@ The `Dataset` base class handles transforms application:
...
 ```python
 class BaseTransform(ABC):
-    def __call__(self, input_dict: Dict[str, Any]) -> Dict[str, Any]:
+    def __call__(
+        self,
+        input_dict: Dict[str, Any],
+        context: PipelineContext | None = None,
+    ) -> Dict[str, Any]:
+        self.context = context
         return self.transform(input_dict)
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply docs/framework/design.md:160]: https://github.com/tier4/autoware-ml/pull/23/commits/63b77304d6facae3c2bb54cbb01b63de388bbb2e

File: autoware_ml/utils/optimizer.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        dictionaries compatible with PyTorch optimizers.
+    """
+    if hasattr(model, "build_optimizer_groups"):
+        named_groups: Mapping[str, Sequence[nn.Parameter]] = model.build_optimizer_groups()
+    else:
+        named_groups = {"default": [p for p in model.parameters() if p.requires_grad]}
+    overrides = materialize_partial_kwargs(dict(group_overrides or {}))
+
+    if set(named_groups) == {"default"} and not overrides.get("default"):
+        return list(named_groups["default"])
```
[the remark this answers is not in the document]
Comment (amadeuszsz) [reply autoware_ml/utils/optimizer.py:107]: https://github.com/tier4/autoware-ml/pull/23/commits/57d8087b4dc343099eb78a1a5123d190a7504b9a
